src/decode.rs

```diff
@@ -466,4 +465,11 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
Great, thanks for this test case. However, and sorry for being picky, decoding &[b'.'] and &[b'.', x] (where x is some byte other than b'.') are not covered yet. Could you provide a test case?

Reply:
yeah that's fine